dialog/text_box.gd

```diff
@@ -0,0 +1,59 @@
+extends MarginContainer
+class_name TextBox
+
+@onready var label = $Text/Label
+@onready var timer: Timer = $CharDisplayTimer
+
+@export_group("TextBox settings")
+@export var max_width: int = 256
+@export var text_speed: float = 0.03
+
+# values based on player sprite
+@export var y_offset: float = 40.0
+@export var x_offset: float = 24.0
+@export_group("")
+
+var text: String = ""
+var char_index: int = 0
+var tracked_node: Node2D
+
+signal finished_displaying()
+
+func _process(delta: float) -> void:
+	global_position = _get_adjusted_global_position()
+
+func _get_adjusted_global_position() -> Vector2:
+	var adjusted_position: Vector2 = tracked_node.global_position
```

Comment:
Where are you getting the tracked_node from?

Reply:
When you invoke the dialog manager method to start a dialog, you pass a reference to the object you want to track

Reply:
[here](https://github.com/zoepfister/maze-secret/blob/dialogs/dialog/DialogManager.gd#L27C1-L33C26) is where we set the tracked object for a given textbox